test/suites/utilization/fips_test.go

```diff
@@ -0,0 +1,239 @@
+	Context("FIPS Validation", func() {
+		It("should reject invalid FIPSMode values", func() {
+			invalidFIPSMode := v1beta1.FIPSMode("InvalidMode")
+			nodeClass.Spec.FIPSMode = &invalidFIPSMode
+
+			Expect(env.Client.Create(env.Context, nodeClass)).ToNot(Succeed())
+		})
+
+		It("should reject FIPS + Ubuntu2204 combination", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.Ubuntu2204ImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr(v1beta1.FIPSModeFIPS)
+
+			Expect(env.Client.Create(env.Context, nodeClass)).ToNot(Succeed())
+		})
+
+		It("should reject FIPS without SIG access", func() {
+			if !env.InClusterController {
+				Skip("Testing FIPS usage cleanly fails without SIG access only makes sense in self-hosted mode - NAP has SIG access")
+			}
+
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr(v1beta1.FIPSModeFIPS)
+
+			env.ExpectCreated(nodeClass, nodePool)
+
+			// Should fail to reconcile images due to SIG requirement
+			Eventually(func(g Gomega) {
+				g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(nodeClass), nodeClass)).To(Succeed())
+				condition := nodeClass.StatusConditions().Get(v1beta1.ConditionTypeImagesReady)
+				g.Expect(condition.IsFalse()).To(BeTrue())
+				g.Expect(condition.Reason).To(Equal("SIGRequiredForFIPS"))
+				g.Expect(condition.Message).To(ContainSubstring("FIPS images require UseSIG to be enabled"))
+
+				// Check that the overall Ready condition is also false
+				readyCondition := nodeClass.StatusConditions().Get(opstatus.ConditionReady)
+				g.Expect(readyCondition.IsFalse()).To(BeTrue())
+			}).Should(Succeed())
+		})
+	})
+
+	Context("FIPS Provisioning", func() {
+		BeforeEach(func() {
+			if env.InClusterController {
+				Skip("FIPS tests require SIG access - skipping in self-hosted mode")
+			}
+		})
+
+		It("should provision FIPS-enabled Ubuntu nodes", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.UbuntuImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr(v1beta1.FIPSModeFIPS)
+
+			pod := test.Pod()
+			env.ExpectCreated(nodeClass, nodePool, pod)
+			env.EventuallyExpectHealthy(pod)
+			node := env.ExpectCreatedNodeCount("==", 1)[0]
+
+			imageRef := validateNodeUsesFIPS(node)
+			validateNodeClassHasExpectedImages(imageRef, true) // true = expect FIPS images
+		})
+
+		It("should provision FIPS-enabled AzureLinux nodes", func() {
+			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			nodeClass.Spec.FIPSMode = lo.ToPtr(v1beta1.FIPSModeFIPS)
+
+			pod := test.Pod()
+			env.ExpectCreated(nodeClass, nodePool, pod)
+			env.EventuallyExpectHealthy(pod)
+			node := env.ExpectCreatedNodeCount("==", 1)[0]
+
+			imageRef := validateNodeUsesFIPS(node)
+			validateNodeClassHasExpectedImages(imageRef, true) // true = expect FIPS images
+		})
+	})
+
+	Context("FIPS Disabled", func() {
+		It("should provision nodes with FIPS explicitly disabled", func() {
```

Comment:
Maybe leave a comment here explaining the significance of explicitly disabled versus the default (which all the other non-FIPS tests are hitting?)